### PR TITLE
141  add futures

### PR DESCRIPTION
**Pull request by rlkelly**

### CODE REVIEW

File: src/accountant.rs [hunk 1]
```diff
@@ -81,6 +81,12 @@ impl Accountant {
         true
     }
 
+    // fn forget_signature_with_last_id(&self, last_id: &Hash) -> bool {
```
Comment (garious) [inline src/accountant.rs]: Don't need this. Even if not commented out, it's not right. It should remove `last_id`, not the last item.

File: src/accountant_stub.rs [hunk 2]
```diff
@@ -51,35 +52,38 @@ impl AccountantStub {
     /// Request the balance of the user holding `pubkey`. This method blocks
     /// until the server sends a response. If the response packet is dropped
     /// by the network, this method will hang indefinitely.
-    pub fn get_balance(&self, pubkey: &PublicKey) -> io::Result<Option<i64>> {
+    pub fn get_balance(&self, pubkey: &PublicKey) -> FutureResult<i64, i64> {
```
Comment (garious) [inline src/accountant_stub.rs]: How about something like`FutureResult<Option<i64>, io::Error>` ? Or perhaps the library offers a `futures::io::FutureResult<Option<i64>>`?

File: src/accountant_stub.rs [hunk 3]
```diff
@@ -51,35 +52,38 @@ impl AccountantStub {
...
         let req = Request::GetBalance { key: *pubkey };
         let data = serialize(&req).expect("serialize GetBalance");
-        self.socket.send_to(&data, &self.addr)?;
+        self.socket.send_to(&data, &self.addr).expect("buffer error");
```
Comment (garious) [inline src/accountant_stub.rs]: Can't panic on this one. It's too likely to happen.

File: src/accountant_stub.rs [hunk 4]
```diff
@@ -51,35 +52,38 @@ impl AccountantStub {
...
-        Ok(None)
+        err(0)
     }
 
     /// Request the last Entry ID from the server. This method blocks
     /// until the server sends a response. At the time of this writing,
     /// it also has the side-effect of causing the server to log any
     /// entries that have been published by the Historian.
-    pub fn get_last_id(&self) -> io::Result<Hash> {
+    pub fn get_last_id(&self) -> FutureResult<Hash, ()> {
```
Comment (garious) [inline src/accountant_stub.rs]: Again, `io::Errror` instead of `()`?

**Comment by rlkelly** [comment]
sure thing, i'll knock this out in a few hours

**Comment by garious** [comment]
See #150

**Pull request closed without merging**